trajopt_optimizers/trajopt_sqp/test/cart_position_optimization_unit.cpp

```diff
@@ -0,0 +1,181 @@
+#include <trajopt_ifopt/variable_sets/joint_position_variable.h>
+#include <trajopt_ifopt/costs/squared_cost.h>
+#include <trajopt_ifopt/utils/trajopt_utils.h>
+
+const bool DEBUG = false;
+
+inline std::string locateResource(const std::string& url)
+{
+  std::string mod_url = url;
+  if (url.find("package://trajopt") == 0)
+  {
+    mod_url.erase(0, strlen("package://trajopt"));
+    size_t pos = mod_url.find('/');
+    if (pos == std::string::npos)
+    {
+      return std::string();
+    }
+
+    std::string package = mod_url.substr(0, pos);
+    mod_url.erase(0, pos);
+    std::string package_path = std::string(TRAJOPT_DIR);
+
+    if (package_path.empty())
+    {
+      return std::string();
+    }
+
+    mod_url = package_path + mod_url;
+  }
+
+  return mod_url;
+}
+
+class CartPositionOptimization : public testing::TestWithParam<const char*>
+{
+public:
+  // 1) Create the problem
+  ifopt::Problem nlp_;
+
+  Eigen::VectorXd joint_target_;
+
+  void SetUp() override
+  {
+    if (DEBUG)
+      console_bridge::setLogLevel(console_bridge::LogLevel::CONSOLE_BRIDGE_LOG_DEBUG);
+    else
+      console_bridge::setLogLevel(console_bridge::LogLevel::CONSOLE_BRIDGE_LOG_NONE);
+
+    // 1)  Load Robot
+    boost::filesystem::path urdf_file(std::string(TRAJOPT_DIR) + "/test/data/arm_around_table.urdf");
+    boost::filesystem::path srdf_file(std::string(TRAJOPT_DIR) + "/test/data/pr2.srdf");
+    tesseract_scene_graph::ResourceLocator::Ptr locator =
+        std::make_shared<tesseract_scene_graph::SimpleResourceLocator>(locateResource);
+    auto tesseract = std::make_shared<tesseract::Tesseract>();
+    tesseract->init(urdf_file, srdf_file, locator);
+
+    // Extract necessary kinematic information
+    auto forward_kinematics = tesseract->getFwdKinematicsManager()->getFwdKinematicSolver("right_arm");
+    tesseract_environment::AdjacencyMap::Ptr adjacency_map = std::make_shared<tesseract_environment::AdjacencyMap>(
+        tesseract->getEnvironment()->getSceneGraph(),
+        forward_kinematics->getActiveLinkNames(),
+        tesseract->getEnvironment()->getCurrentState()->link_transforms);
+    auto kinematic_info = std::make_shared<trajopt::CartPosKinematicInfo>(
+        forward_kinematics, adjacency_map, Eigen::Isometry3d::Identity(), forward_kinematics->getTipLinkName());
+
+    // Get target position
+    Eigen::VectorXd start_pos(forward_kinematics->numJoints());
+    start_pos << 0.0, 0, 0, -1.0, 0, -1, -0.00;
+    if (DEBUG)
+      std::cout << "Joint Limits:\n" << forward_kinematics->getLimits().transpose() << std::endl;
+
+    auto target_pose = Eigen::Isometry3d::Identity();
+    joint_target_ = start_pos;
+    forward_kinematics->calcFwdKin(target_pose, joint_target_);
+    //    auto world_to_base =
+    //        tesseract->getEnvironment()->getCurrentState()->link_transforms.at(forward_kinematics->getBaseLinkName());
+    //    target_pose = world_to_base * target_pose;
```

Comment:
should this be removed?

Reply:
So the test fails if I uncomment this. Perhaps that means something is wrong. I think this test is copied from trajopt. world_to_base does have a translation.

Reply:
@Levi-Armstrong Could you look at the cartesian constraint and see if anything looks wrong? It is basically the same as whats in CartPoseErrCalculator::operator()

Reply:
I created a PR against your branch fixing the unit test.